nucypher/policy/policies.py

```diff
@@ -729,120 +614,72 @@ def generate_policy_parameters(n: int,
 
         # Transact  # TODO: Move this logic to BlockchainPolicyActor
-        receipt = self.author.policy_agent.create_policy(
-            policy_id=self.hrac(),  # bytes16 _policyID
-            author_address=self.author.checksum_address,
+        receipt = self.alice.policy_agent.create_policy(
+            policy_id=self.hrac,  # bytes16 _policyID
+            author_address=self.alice.checksum_address,
             value=self.value,
             end_timestamp=self.expiration.epoch,  # uint16 _numberOfPeriods
-            node_addresses=prearranged_ursulas  # address[] memory _nodes
+            node_addresses=addresses  # address[] memory _nodes
         )
 
         # Capture Response
-        self.receipt = receipt
-        self.publish_transaction = receipt['transactionHash']
-        self.is_published = True  # TODO: For real: TX / Swarm confirmations needed?
+        return receipt['transactionHash']
 
-        return receipt
+    def _blockchain_signer(self):
+        transacting_power = self.alice._crypto_power.power_ups(TransactingPower)
+        return transacting_power.sign_message
 
-    def make_arrangement(self, ursula: Ursula, *args, **kwargs):
-        return self._arrangement_class(alice=self.alice,
-                                       expiration=self.expiration,
-                                       ursula=ursula,
-                                       rate=self.rate,
-                                       duration_periods=self.duration_periods,
-                                       *args, **kwargs)
+    def _make_enactment_payload(self, publication_transaction, kfrag):
+        return bytes(publication_transaction) + bytes(kfrag)
 
-    def enact(self, network_middleware, publish_to_blockchain=True, publish_treasure_map=True) -> NodeEngagementMutex:
-        """
-        Assign kfrags to ursulas_on_network, and distribute them via REST,
-        populating enacted_arrangements
-        """
-        if publish_to_blockchain is True:
-            self.publish_to_blockchain()
+    def _enact_arrangements(self,
+                            network_middleware,
+                            arrangements,
+                            publish_treasure_map=True) -> TreasureMapPublisher:
+        transaction = self._publish_to_blockchain(list(arrangements))
+        return super()._enact_arrangements(network_middleware=network_middleware,
+                                           arrangements=arrangements,
+                                           publish_treasure_map=publish_treasure_map,
+                                           publication_transaction=transaction)
 
-            # Not in love with this block here, but I want 121 closed.
-            for arrangement in self._accepted_arrangements:
-                arrangement.publish_transaction = self.publish_transaction
 
-        publisher = super().enact(network_middleware, publish_treasure_map=False)
+class EnactedPolicy:
```

Comment:
Trying to familiarize with the new object introduced in this PR.  Enacted policies can be federated or decentralized?

Reply:
They can be both, both `FederatedPolicy.enact()` and `BlockchainPolicy.enact()` return an `EnactedPolicy` (well, it's `Policy.enact()` for both). The contents of `EnactedPolicy` are essentially based on how the policy object was used after `enact()`. It is possible they can be grouped differently.